clang/lib/Driver/ToolChains/HLSL.cpp

```diff
@@ -35,6 +38,11 @@ bool isLegalShaderModel(Triple &T) {
   if (Version.getSubminor())
     return false;
 
+  // DXIL Version should be set as SubArch
+  if (T.getSubArch() == Triple::NoSubArch) {
+    return false;
+  }
```

Comment:
nit:
```suggestion
  if (T.getSubArch() == Triple::NoSubArch)
    return false;
```

see: https://llvm.org/docs/CodingStandards.html#don-t-use-braces-on-simple-single-statement-bodies-of-if-else-loop-statements